src/Channels.Networking.Libuv/UvTcpConnection.cs

```diff
@@ -185,10 +185,10 @@ private static Uv.uv_buf_t AllocCallback(UvStreamHandle handle, int status, obje
             return ((UvTcpConnection)state).OnAlloc(handle, status);
         }
 
-        private Uv.uv_buf_t OnAlloc(UvStreamHandle handle, int status)
+        private unsafe Uv.uv_buf_t OnAlloc(UvStreamHandle handle, int status)
         {
             _inputBuffer = _input.Alloc(2048);
-            return handle.Libuv.buf_init(_inputBuffer.Memory.BufferPtr, _inputBuffer.Memory.Length);
+            return handle.Libuv.buf_init((IntPtr)_inputBuffer.Memory.UnsafePointer, _inputBuffer.Memory.Length);
```

Comment:
To do it safely, you need to first call TryGetArray. If it succeeds, pin the array to get the pointer. Otherwise, use UnsafeArray (as it's apparently native memory).